astropy/coordinates/sky_coordinate.py

```diff
@@ -833,6 +833,13 @@ def to_string(self, style='decimal', **kwargs):
         lonargs.update(kwargs)
         latargs.update(kwargs)
 
+        if style == 'hmsdms' :  # for hmsdms, want to adjust precision of RA to be one more than 
+                                # that or dec, because hours are 15 * cos(dec) bigger than degrees.
+            # default case; precision = 0 for both
+            if not latargs.has_key('precision') : lonargs['precision'] = 1
```

Comment:
`has_key` has been deprecated for a while and was removed in Python 3.x.  The more recent way to do this is ``if 'precision' not in latargs:``

Also, a minor style point: generally PEP8 (the Python style guide) favors having even one-line if blocks be in their own line, so that it's clearer from the indentation what the logical flow is.